smart-contract/assembly/contracts/internals/chunks.ts

```diff
@@ -1,120 +1,124 @@
-import { stringToBytes, bytesToU32, u32ToBytes } from '@massalabs/as-types';
+import { stringToBytes, u32ToBytes } from '@massalabs/as-types';
 import { sha256, Storage } from '@massalabs/massa-as-sdk';
-import { CHUNK_NB_TAG, FILE_TAG, CHUNK_TAG } from './const';
-import { _removeFilePath } from './file-list';
+import { fileChunkKey } from './storageKeys/chunksKeys';
+import { bytesToU32 } from '@massalabs/as-types';
+import { fileChunkCountKey } from './storageKeys/chunksKeys';
+import { _removeFileLocation } from './location';
+import { _removeAllFileMetadata } from './metadata';
+import { FILE_TAG, CHUNK_TAG } from './storageKeys/tags';
+import { fileMetadataLocationKey } from './storageKeys/metadataKeys';
+/* -------------------------------------------------------------------------- */
+/*                                     SET                                    */
+/* -------------------------------------------------------------------------- */
 
 /**
  * Sets a chunk of a file in storage.
- * @param filePath - The path of the file.
+ * @param location - The location of the file.
  * @param index - The index of the chunk.
  * @param chunk - The chunk data.
- * @throws If the chunk index is not in the expected order.
+ * @throws If the total chunks weren't set for this file or if the index is out of bounds.
  */
 export function _setFileChunk(
-  filePath: string,
+  location: string,
   index: u32,
   chunk: StaticArray<u8>,
 ): void {
-  const filePathHash = sha256(stringToBytes(filePath));
-  const totalChunks = _getTotalChunk(filePathHash);
+  const hashLocation = sha256(stringToBytes(location));
+  const totalChunk = _getTotalChunk(hashLocation);
 
-  assert(totalChunks > 0, "Total chunks wasn't set for this file");
-  assert(index < totalChunks, 'Index out of bounds');
+  assert(totalChunk > 0, "Total chunks wasn't set for this file");
+  assert(index < totalChunk, 'Index out of bounds');
 
-  Storage.set(_getChunkKey(filePathHash, index), chunk);
+  Storage.set(fileChunkKey(hashLocation, index), chunk);
 }
 
-export function _removeChunksRange(
-  filePathHash: StaticArray<u8>,
-  start: u32 = 0,
-  end: u32 = 0,
-): void {
-  for (let i = u32(start); i < end; i++) {
-    Storage.del(_getChunkKey(filePathHash, i));
-  }
-}
-
-export function _setTotalChunk(
-  filePathHash: StaticArray<u8>,
-  totalChunks: u32,
-): void {
-  Storage.set(_getTotalChunkKey(filePathHash), u32ToBytes(totalChunks));
-}
+/* -------------------------------------------------------------------------- */
```

Comment:
maybe also add a delete_chunks function like documented here https://forum.massa.community/t/community-feedback-proposal-for-improving-website-storage-on-deweb/82/15#p-319-smart-contract-code-2 for more client-side flexibility and future-proofing